src/dune_rules/dune_file.ml

```diff
@@ -288,47 +346,36 @@ module Script = struct
     Path.build generated_dune_file
     |> Io.Untracked.with_lexbuf_from_file ~f:(Dune_lang.Parser.parse ~mode:Many)
     |> List.rev_append from_parent
-    |> parse ~dir ~file:(Some file) ~project
+    |> parse ~mask ~dir ~file:(Some file) ~project
   ;;
 end
 
-let filter_out_stanzas_from_hidden_packages ~visible_pkgs =
-  List.filter_map ~f:(fun stanza ->
-    let include_stanza =
-      match Stanzas.stanza_package stanza with
-      | None -> true
-      | Some package ->
-        let name = Package.name package in
-        Package.Name.Map.mem visible_pkgs name
-    in
-    if include_stanza
-    then Some stanza
-    else (
-      match Stanza.repr stanza with
-      | Library.T l ->
-        Library_redirect.Local.of_private_lib l
-        |> Option.map ~f:Library_redirect.Local.make_stanza
-      | _ -> None))
-;;
-
-let filter_stanzas (mask : Only_packages.t) (dune_files : t list) =
-  match mask with
-  | None -> dune_files
-  | Some visible_pkgs ->
-    List.map dune_files ~f:(fun dune_file ->
-      { dune_file with
-        stanzas = filter_out_stanzas_from_hidden_packages ~visible_pkgs dune_file.stanzas
-      })
+let check_dynamic_stanza =
+  (* CR-rgrinberg: unfortunately this needs to kept in sync with the rules
+     manually *)
```

Comment:
What about making this boolean flag part of the definition of the stanza (ie an argument of `Stanza.Make`)? That way the programmer is forced to provide this logic when adding new stanzas.

Reply:
I tried this, but it ends up being a pretty massive change. I do agree that something like this would be good and we should continue using `Stanza.Make` to make the stanza interface more uniform. Will leave a TODO for this.